src/platform/forms-system/src/js/utilities/file/checkTypeAndExtensionMatches.js

```diff
@@ -89,13 +89,16 @@ export default function checkTypeAndExtensionMatches({ file, result }) {
   const match = extension && fileTypeSignatures[extension];
   const signature = match?.sig;
 
-  if (signature && match?.mime === file.type) {
-    return arrayIncludesArray(
-      result,
-      Array.isArray(signature)
-        ? signature
-        : [...signature].map(str => str.charCodeAt(0)),
-    );
+  if (typeof signature !== 'undefined' && match?.mime === file.type) {
+    if (extension === 'txt') return true;
+    else {
+      return arrayIncludesArray(
+        result,
+        Array.isArray(signature)
+          ? signature
+          : [...signature].map(str => str.charCodeAt(0)),
+      );
+    }
```

Comment:
What is `typeof signature !== 'undefined'` doing for us that the truthy check wasn't covering?

Also, could we get a unit test for this added in here?

https://github.com/department-of-veterans-affairs/vets-website/blob/63d0930f8e22a41288d8176edaf3e6353f280883/src/platform/forms-system/test/js/utilities/file.unit.spec.js#L78

Reply:
That check wasn't doing what I thought it was (fixing the problem uploading .png files) so I've removed it and reworked the `extension` check slightly.  Also I've added a unit test to that file.